README.md

```diff
@@ -132,6 +132,89 @@ is not supported anymore. Exporting `$ROLENAME_provider_os_default` allows the u
+ To fix, either use a properly idempotent module to obtain the information (e.g. instead of
+ using `command: cat` to read file into a registered variable, use `slurp` and apply `.content|b64decode`
+ to the result like [here](https://github.com/linux-system-roles/kdump/pull/23/files#diff-d2414d4ec8ba189e1a244b0afc9aa81eL8)), or apply proper `check_mode:` and `changed_when:`
+ attributes to the task. [more_info](https://github.com/ansible/molecule/issues/128#issue-135906202).
+
+* Another problem are commands that you need to execute to make changes. In check mode, you
+ need to test for changes without actually applying them. If the command has some kind of "--dry-run"
+ flag to enable executing without making actual changes, use it in check_mode (use the variable
+ `ansible_check_mode` to determine whether we are in check mode). But you then need to set `changed_when:`
+ according to the command status or output to indicate changes. See (https://github.com/linux-system-roles/selinux/pull/38/files#diff-2444ad0870f91f17ca6c2a5e96b26823L101) for an example.
+
+* Another problem is using commands that get installed during the install phase, which is
+ skipped in check mode. This will make check mode fail if the role has not been executed
+ before (and the packages are not there), but DTRT if check mode is executed after normal mode. 
+
+* To view reasoning for supporting why check mode in first execution may not be worthwhile: see [here](https://github.com/ansible/molecule/issues/128#issuecomment-245009843). If this is to be supported, see hhaniel's proposal [here](https://github.com/linux-system-roles/timesync/issues/27#issuecomment-472466223), which seems to properly guard even against such cases.
+
+## Supporting multiple distributions and versions 
+
+If some tasks need to be parameterized according to distribution and version (name of packages,
+configuration file paths, names of services), use this in the beginning of your `tasks/main.yml`:
+```yaml
+- name: Set version specific variables
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution }}-{{ ansible_distribution_version }}.yml"
+    - "{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
+    - "{{ ansible_distribution }}.yml"
+    - "{{ ansible_os_family }}.yml"
+```
+
+You can add "-default.yml" at the end to set the variables for unrecognized distributions from  a common default.yml file. If this is not done, the role will fail for unrecognized distribution (this may or may not be intended). Put the distribution-specific variables into appropriate files under "vars/". Unfortunately, this will lead to duplicate vars files for similiar distibutions (e.g. CentOS 7 and RHEL 7). In cases such as these, use symlinks. 
+
+```If the user is on Windows and symlinks are not working, have them request the extension or point the user WSL to Galaxy to resolve symlink issue for Windows users. https://github.com/linux-system-roles/network/pull/64```
```

Comment:
I would remove this whole sentence. I since then found out that Ansible is not supported on Windows at all and the reporter of that issue did not reply when I asked why was it needed.